crates/rspack_core/src/chunk_graph/chunk_graph_chunk.rs

```diff
@@ -674,6 +697,14 @@ impl ChunkGraph {
       ) {
         for connection in module_graph.get_outgoing_connections(&module) {
           // https://github.com/webpack/webpack/blob/1f99ad6367f2b8a6ef17cce0e058f7a67fb7db18/lib/ChunkGraph.js#L290
+          if set.contains(connection.module_identifier()) {
+            let maybe_transitive = module_graph
+              .dependency_by_id(&connection.dependency_id)
+              .is_some_and(|dep| matches!(dep.dependency_type(), DependencyType::ExtractCSS));
```

Comment:
How's the benefit here, if improve the perf a lot I'm ok with this, but if not I don't recommend to depend on `DependencyType::ExtractCSS` here